sklearn/feature_extraction/_hashing.pyx

```diff
@@ -45,7 +43,7 @@ def transform(raw_X, Py_ssize_t n_features, dtype):
 
     for x in raw_X:
         for f, v in x:
-            if isinstance(v, string_types):
+            if isinstance(v, basestring):
```

Comment:
Hmm I'd prefer using (str, unicode). We don't have published "minimum cython requirements" but I suppose there's no harm in this case in maintaining backwards compatibility?